pkg/vulnsrc/vulnsrc.go

```diff
@@ -61,53 +64,89 @@ func init() {
 	}
 }
 
-func Update(targets []string, cacheDir string, light bool, updateInterval time.Duration) error {
+type Operation interface {
+	SetMetadata(db.Metadata) error
+}
+
+type Updater struct {
+	dbc            Operation
+	updateMap      map[string]VulnSrc
+	cacheDir       string
+	dbType         db.Type
+	updateInterval time.Duration
+	clock          clock.Clock
+	optimizer      Optimizer
+}
+
+func NewUpdater(cacheDir string, light bool, interval time.Duration) Updater {
```

Comment:
Comment: Today this returns an `Updater`, as far as I can see there are no functions of it that do any modifications to any of the `Updater` struct members. Tomorrow this might change and we'll need to return `*Updater` to have those modifications stick.

Reply:
But in the spirit of keeping things simple, I'd leave it as is for now. Just wanted to make an observation.